echopype/convert/parse_base.py

```diff
@@ -89,13 +89,26 @@ def rectangularize_data(self):
                     self.ch_ids[data_type].append(k)
                     self.ping_data_dict[data_type][k] = self.pad_shorter_ping(v)
             # Transmit data
-            for k, v in self.ping_data_dict_tx[data_type].items():
-                if all(
-                    (x is None) or (x.size == 0) for x in v
-                ):  # if no data in a particular channel
-                    self.ping_data_dict_tx[data_type][k] = None
-                else:
-                    self.ping_data_dict_tx[data_type][k] = self.pad_shorter_ping(v)
+            self.rectangularize_transmit_ping_data(data_type)
```

Comment:
I think it is better to remove this call of `rectangularize_transmit_ping_data` from this function, and change your current call of `rectangularize_data` to 2 calls: `rectangularize_data` and `rectangularize_transmit_ping_data`. When I read that section in `echopype/convert/api.py` I was wondering where the `rectangularize_transmit_ping_data` call went for some of the cases and that was confusing.

Reply:
As we discussed, we will not change this for now.